### PR TITLE
feat(lb-pn): add wait for attachement to load balancer

### PR DESCRIPTION
**Pull request by Monitob**
- Enable wait a stable status when private network is attached to a load balancer

### CODE REVIEW

File: api/lb/v1/lb_utils.go [hunk 1]
```diff
@@ -87,3 +87,70 @@ func waitForLb(timeout *time.Duration, retryInterval *time.Duration, getLB func(
...
+	if timeout != nil {
+		t = *timeout
+	}
+	r := defaultRetryInterval
+	if retryInterval != nil {
+		r = *retryInterval
+	}
+
+	terminalStatus := map[PrivateNetworkStatus]struct{}{
+		PrivateNetworkStatusReady: {},
```
Comment (kindermoumoute) [inline api/lb/v1/lb_utils.go:111]: what about unknown?

File: api/lb/v1/lb_utils.go [hunk 2]
```diff
@@ -87,3 +87,70 @@ func waitForLb(timeout *time.Duration, retryInterval *time.Duration, getLB func(
 	}
 	return lb.(*LB), nil
 }
+
+// ZonedAPIWaitForLBPNRequest is used by WaitForLBPN method.
+type ZonedAPIWaitForLBPNRequest struct {
+	LBID          string
+	PNID          string
```
Comment (kindermoumoute) [inline api/lb/v1/lb_utils.go:94]: Why do you need to wait for PN one by one? Why not wait for all PNs to be up everytime?

Comment (Monitob) [reply api/lb/v1/lb_utils.go:94]: Today we only support one pn but I can adapt this for future upgrades

File: api/lb/v1/lb_utils.go (hunk 1 above)
Comment (Monitob) [reply api/lb/v1/lb_utils.go:111]: Not a stable state

**Review by remyleone — APPROVED** [review]

**Pull request merged**